shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -826,7 +826,9 @@ - (void)applicationBecameActive:(NSNotification*)notification {
 
 - (void)applicationWillResignActive:(NSNotification*)notification {
   TRACE_EVENT0("flutter", "applicationWillResignActive");
-  self.view.accessibilityElementsHidden = YES;
+  if ([FlutterViewController isUIAccessibilityIsVoiceOverRunning]) {
+    self.view.accessibilityElementsHidden = YES;
```

Comment:
What about when someone is using a physical switch device or voice input? Does this break things in those cases?

Reply:
/cc @chunhtai

Reply:
this looks like it will break if voice control is turned on, unfortunately there isn't an API to detect whether voice control is on last time i checked

Reply:
Could we flip the sense of this and allow tests to pass some parameter/set some environment variable to override this behavior?

Reply:
Do we need to `self.view.accessibilityElementsHidden = YES;` if voice input is on? I remember `self.view.accessibilityElementsHidden = YES;` is to fix the exact issue where the voice over is on and the focus is jumped to flutter view.

Reply:
So in the current code, we always hide accessibility elements. Now, we only hide the elements when voice over is turned on. This would actually be an improvement for voice input (by not hiding a11y elements when voice over is off)

I couldn't find any concrete example for voice input to test tho.

Reply:
@chunhtai @dnfield Friendly ping. Did my comment make sense? I'm happy to schedule a quick VC to discuss this :)

Reply:
I think it's probably fine, but it would be good to test an app that has voice input turned on while it shows a permissions dialog, and make sure you can use voice input there the same as you would be able to use it in a regular native iOS app in that situation. I'm not familiar enough with voice input to know how to test this myself.

If a manual test shows that it works ok, that's probably the best we can do. If not then we need to figure out how to change this only for tests.